src/solver/td3.ml

```diff
@@ -340,7 +377,34 @@ module Base =
             | _ ->
               (* The RHS is re-evaluated, all deps are re-trigerred *)
               HM.replace dep x VS.empty;
-              eq x (eval l x) (side ~x)
+              if narrow_globs then
+                let acc = HM.create 0 in
+                let changed = HM.create 0 in
+                Fun.protect ~finally:(fun () -> (
+                      if narrow_globs_eliminate_dead then begin
+                        let prev_sides_x = HM.find_option prev_sides x in
+                        Option.may (VS.iter (fun y ->
+                            if not @@ HM.mem acc y then begin
+                              ignore @@ divided_side D_Narrow x y (S.Dom.bot ());
+                              if S.Dom.is_bot @@ HM.find rho y then
+                                let casualties = S.postmortem y in
+                                List.iter (fun x -> solve x Widen) casualties
```

Comment:
This has similar connotations to #365: calling `solve` like this on an `x` which is influenced by the current `y` turns it into a forward-propagating solver (in part), as opposed to remaining fully demand-driven.
Like with #365, this seems like it could ask to solve things which maybe nothing depends on any more, just doing extra work.

Reply:
This is precisely why that is there! I'll share the draft paper with you that explains this once we have a little more time next week.

Reply:
This is closely tied to the remark above: The idea is that for those unknowns in `postmortem` we are willing to pay this additional penalty in order to collect dead side-effects and potentially improve the values at globals. Dead code will be pruned at the end by `reach` but that doesn't shrink side-effects.

For those with access to our drafts at https://github.com/tum-cit-pl, this is discussed in https://github.com/tum-cit-pl/combatting-precision-loss Section 4.3, where we also have the remark that forward propagating solvers exhibit this behavior out of the box.